ipapython/ipautil.py

```diff
@@ -173,8 +173,13 @@ def __init__(self, addr, match_local=False, parse_netmask=True,
             iface = None
             for interface in netifaces.interfaces():
                 for ifdata in netifaces.ifaddresses(interface).get(family, []):
+
+                    # link-local addresses contain '%suffix' that causes parse
+                    # errors in IPNetwork
+                    addr = ifdata['addr'].split(u'%', 1)[0]
```

Comment:
The variable `addr` shadows another variable from the same namespace. This results in an incorrect IP address in the error message when the IP does not mach any network interface.

```
# ipa-client-install --ip-address '2201:db8::'
...
no network interface matches the IP address and netmask fe80::21a:4aff:fe23:1530
```